### PR TITLE
dvc-referrer headers, deprecated function cleanup

### PR DESCRIPTION
**Pull request by aurorawalker**

### CODE REVIEW

File: internal/dvc_oauth/auth.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func GetAuthToken(clientId, clientSecret string) (Auth0, error) {
 	res, _ := http.DefaultClient.Do(req)
 
 	defer res.Body.Close()
-	body, _ := ioutil.ReadAll(res.Body)
+	body, _ := io.ReadAll(res.Body)
```
Comment (aurorawalker) [inline internal/dvc_oauth/auth.go:28]: IDE gave a deprecation warning on this, documentation indicated that this was a clear 1:1 swap. See https://pkg.go.dev/io/ioutil#ReadAll

File: internal/provider/provider.go [hunk 2]
```diff
@@ -3,14 +3,14 @@ package provider
...
 	"fmt"
-	"github.com/hashicorp/terraform-provider-scaffolding-framework/internal/dvc_oauth"
-	"os"
-
 	dvc_mgmt "github.com/devcyclehq/go-mgmt-sdk"
 	dvc_server "github.com/devcyclehq/go-server-sdk"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-scaffolding-framework/internal/dvc_oauth"
```
Comment (aurorawalker) [inline internal/provider/provider.go:11]: Whoop - I think this was IDE autoformatting to alphabetical. Can undo if desired.

**Comment by aurorawalker** [comment]
@JamieSinn  Still uncertain about these automated tests failing (and struggling to get the tests to run correctly locally with the API keys - not clear which organization/project it has to be?) so some extra guidance would be appreciated!

File: docs/data-sources/environment.md [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
 subcategory: ""
 description: |-
-  DevCycle Environment Data Source.
+  DevCycle Environment Data Source. Read data from a given DevCycle Environment. R
 ---
 
 # devcycle_environment (Data Source)
 
-DevCycle Environment Data Source.
+DevCycle Environment Data Source. Read data from a given DevCycle Environment. R
```
Comment (aurorawalker) [inline docs/data-sources/environment.md:11]: All of these changes to the .md files were the result of running the `go generate` command

**Comment by JamieSinn** [comment]
The errors you're getting are from the client tests; they have a line number in the failure to tell you where it failed. You should be able to get a better understanding of what the error actually is from that.

File: internal/dvc_oauth/auth.go (hunk 1 above)
Comment (JamieSinn) [reply internal/dvc_oauth/auth.go:28]: yep - that's fine

File: internal/provider/provider.go (hunk 2 above)
Comment (JamieSinn) [reply internal/provider/provider.go:11]: Also fine - depends on which IDE you use that does or doesn't use this gofmt opt

File: internal/provider/provider.go [hunk 4]
```diff
@@ -81,6 +81,9 @@ func (p *provider) Configure(ctx context.Context, req tfsdk.ConfigureProviderReq
 
 	config := dvc_mgmt.NewConfiguration()
 	config.AddDefaultHeader("Authorization", p.AccessToken)
+	config.AddDefaultHeader("dvc-referrer", "terraform")
+	metadata := fmt.Sprintf("{\"go_version\": \"%s\", \"terraform_version\": \"%s\"}", runtime.Version(), req.TerraformVersion)
```
Comment (JamieSinn) [inline internal/provider/provider.go:85]: Go version here may not be needed - as these terraform modules/providers are provided in a compiled binary context. The better facet here would be the provider version.

Comment (aurorawalker) [reply internal/provider/provider.go:85]: I will consult with the right people - this was explicitly given on the ticket to be the data to collect, but I'll check the reasoning and share your recommendation!

File: internal/provider/provider.go [hunk 5]
```diff
@@ -81,6 +80,9 @@ func (p *provider) Configure(ctx context.Context, req tfsdk.ConfigureProviderReq
 
 	config := dvc_mgmt.NewConfiguration()
 	config.AddDefaultHeader("Authorization", p.AccessToken)
+	config.AddDefaultHeader("dvc-referrer", "terraform")
+	metadata := fmt.Sprintf("{\"dvc_terraform_provider_version\": \"%s\", \"terraform_version\": \"%s\"}", p.version, req.TerraformVersion)
```
Comment (kaushalkapasi) [inline internal/provider/provider.go:84]: @JamieSinn PR looks good to me, can you verify these two versions are being pulled the right way? :)

Comment (JamieSinn) [reply internal/provider/provider.go:84]: yep - that's correct - but we need to ensure that the acceptance tests are passing correctly. 
We may need to re-build the test org?

**Comment by aurorawalker** [comment]
@JamieSinn  - @kaushalkapasi  and I have started poking into the tests to try and fix them, but it seems to be a larger issue and far out of scope of our ticket work. Would you be comfortable merging and releasing our changes without the Provider Acceptance Tests succeeding? (I believe we will need you to do this as we lack permission on the repo to do it ourselves).

File: internal/provider/provider.go (hunk 5 above)
Comment (kaushalkapasi) [reply internal/provider/provider.go:87]: Updated the `BasePath` so it actually hits the Bucketing API correctly... how do things look now? 👀

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**